nativelib/src/main/scala/scala/scalanative/runtime/NativeExecutionContext.scala

```diff
@@ -27,21 +58,34 @@ object NativeExecutionContext {
 
     override def reportFailure(t: Throwable): Unit = t.printStackTrace()
 
-    def hasNextTask: Boolean = !queue.isEmpty
-    def availableTasks: Int = queue.size
+    override def stealWork(maxSteals: Int): Unit = {
+      if (maxSteals <= 0) ()
+      else {
+        var steals = 0
+        while (steals < maxSteals && hasAvailableTasks) {
+          doStealWork()
+          steals += 1
+        }
+      }
+    }
 
-    def executeNextTask(): Unit = if (hasNextTask) {
-      queue.dequeue() match {
-        case null => ()
-        case runnable =>
-          try runnable.run()
-          catch { case t: Throwable => QueueExecutionContext.reportFailure(t) }
+    override def stealWork(timeout: FiniteDuration): Unit = {
+      val deadline = System.currentTimeMillis() + timeout.toMillis
+      while (System.currentTimeMillis() < deadline && hasAvailableTasks) {
```

Comment:
Nitpick, but what do you think of reordering the operations here, to avoid a system call when the queue is empty?

```suggestion
      while (hasAvailableTasks && System.currentTimeMillis() < deadline) {
```